### DISCUSSION TITLE
Newby: I need already working sample for study BLE

### DISCUSSION THREAD

**Discussion by pinkfloyd11** [shinyorg/shiny#1320] [Q&A]
Hi at all,

I'm searching working solution sample of BLE client app.

This app should:
1. scan nearby devices
2. pair with one of it
3. receive and send text message

I need one already working because I'm just begin to study MAUI app and since I not have expperience on MAUI enviroment also simply task become difficult.
So starting from one already working and study it is best way for me.

I have also clone repository and see that there are samples but not have understand how to run it's

Sorry for my basic request but this is my level now :-)

Giuseppe

ps.
I have also acquired udemy course: https://1drv.ms/u/s!AmtSTmmIQW6JnaFFBaY5oO6SQp8L6Q?e=naloCc
but when compile I have this error:

System.MissingMethodException: 'Method not found: !!0 Shiny.AndroidExtensions.GetSystemService<!0>(Shiny.AndroidPlatform,string)'

Can someone download and check code please?

**Comment by aritchie** [comment]
@pinkfloyd11 This is not in the spirit of what the forums are for.  I'm happy to see you're studying code and MAUI, but if you really want to learn, maybe spend more time on the underlying platforms first.  There are samples listed in the readme of this project (including BLE) and in the docs.  

Note that I am the author of Shiny and I did not write or have anything to do with this udemy course.  The course used an alpha version of the library and the API surface has changed a great deal since that time.  It also seems they have you abstracting an abstraction which is only going to increase the complexity and make things harder to understand.  

If you don't want to take my suggestions and just want to use the library - then start with the template (also listed in the readme) and use the samples as a guide.

**Comment by pinkfloyd11** [comment]
Hi Ritchie
Thanks for your reply!
Since i love understand what i'm doing is there documentation or tutorial of your framework?
As said before i'm absolute newby and anything seem difficult ( also run your sample 😁😁)

**Reply by aritchie** [reply]
You're honestly going about this the wrong way.  You're looking for someone to give you answers instead of searching for them yourself.  The source code is all open for you to learn and understand.  The readme alone in this repository has links to OTHER samples as well as a link to our docs which I'm working on.

I'm here to answer basic or quick questions about the architecture, not to be an educator.

**Reply by pinkfloyd11** [reply]
For be honest,
 if I'm asking documentation for understand better what I'm doing means that I'm in search of material to study..

But have a theacher is a good idea and of course I will pay for this. 
So also this second option could be usefull
